include/simdjson/dom/serialization.h

```diff
@@ -0,0 +1,210 @@
+  /** Start an array, prints } **/
+  simdjson_really_inline void end_object();
+  /** Prints a true **/
+  simdjson_really_inline void true_atom();
+  /** Prints a false **/
+  simdjson_really_inline void false_atom();
+  /** Prints a null **/
+  simdjson_really_inline void null_atom();
+  /** Prints a number **/
+  simdjson_really_inline void number(int64_t x);
+  /** Prints a number **/
+  simdjson_really_inline void number(uint64_t x);
+  /** Prints a number **/
+  simdjson_really_inline void number(double x);
+  /** Prints a key (string + colon) **/
+  simdjson_really_inline void key(std::string_view unescaped);
+  /** Prints a string. The string is escaped as needed. **/
+  simdjson_really_inline void string(std::string_view unescaped);
+  /** Clears out the content. **/
+  simdjson_really_inline void clear();
+  /** 
+   * Get access to the buffer, it is own by the instance, but
+   * the user can make a copy. 
+   **/
+  simdjson_really_inline std::string_view str() const;
+
+private:
+  // implementation details (subject to change)
+  /** Prints a C string **/
+  simdjson_really_inline void c_str(const char *c);
+  /** Prints one character **/
+  simdjson_really_inline void one_char(char c);
+  /** Backing buffer **/
+  std::vector<char> buffer{}; // not ideal!
+};
+} // namespace dom
+
+/**
+ * Converts JSON to a string.
+ *
+ *   dom::parser parser;
+ *   element doc = parser.parse("   [ 1 , 2 , 3 ] "_padded);
+ *   cout << to_string(doc) << endl; // prints [1,2,3]
+ *
+ */
+template <class T> 
+std::string to_string(T x)   {
+    // in C++, to_string is standard: http://www.cplusplus.com/reference/string/to_string/
+    // Currently minify and to_string are identical but in the future, they may 
+    // differ.
+    simdjson::dom::string_builder<> sb;
+    sb.append(x);
+    std::string_view answer = sb.str();
+    return std::string(answer.data(), answer.size());
+}
+#if SIMDJSON_EXCEPTIONS
+template <class T> 
+std::string to_string(simdjson_result<T> x) {
+    if (x.error()) { throw simdjson_error(x.error()); }
+    return to_string(x.value());
+}
+#endif 
+
+/**
+ * Minifies a JSON element or document, printing the smallest possible valid JSON.
+ *
+ *   dom::parser parser;
+ *   element doc = parser.parse("   [ 1 , 2 , 3 ] "_padded);
+ *   cout << minify(doc) << endl; // prints [1,2,3]
+ *
+ */
+template <class T> 
+std::string minify(T x)  {
+  return to_string(x);
+}
+
+#if SIMDJSON_EXCEPTIONS
+template <class T> 
+std::string minify(simdjson_result<T> x) {
+    if (x.error()) { throw simdjson_error(x.error()); }
+    return to_string(x.value());
+}
+#endif 
+
+
+} // namespace simdjson
+
+
+/**
+ * Print JSON to an output stream.
+ *
+ * @param out The output stream.
+ * @param value The element.
+ * @throw if there is an error with the underlying output stream. simdjson itself will not throw.
+ */
+inline std::ostream& operator<<(std::ostream& out, simdjson::dom::element value) { 
```

Comment:
Note: if you put this inside the simdjson::dom namespace, C++ will find them automatically because it looks for methods in the namespaces of the first two operands; and it will avoid putting stuff in the top level namespace. Not that it matters *that* much.

Reply:
Done.